frontend/src/app/home/services/main.service.ts

```diff
@@ -0,0 +1,84 @@
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
+import { Desk } from './../model/desk.model';
+import { Office } from './../model/office.model';
+import { Printer } from '../model/printer.model';
+import { Room } from '../model/room.model';
+import { Kitchen } from '../model/kitchen.model';
+import { Guest } from '../model/guest.model';
+import { DeskService } from './desk.service';
+import { PrinterService } from './printer.service';
+import { GuestService } from './guest.service';
+import { KitchenService } from './kitchen.service';
+import { OfficeService } from './office.service';
+import { RoomService } from './room.service';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class MainService {
+
+  private select = new Subject<string[]>();
+
+  desk = [];
+  office = [];
+  printer = [];
+  room = [];
+  kitchen = [];
+  guest = [];
+  all = [];
+
+  select$ = this.select.asObservable();
+
+  constructor(private http: HttpClient, deskService: DeskService,
+    officeService: OfficeService, guestService: GuestService,
+    kitchenService: KitchenService, printerService: PrinterService,
+    roomService: RoomService) {
+
+
+    deskService.getDesk().subscribe(data => {
+      data.forEach(x => this.desk.push(new Desk(x)));
+    });
+    officeService.getOffice().subscribe(data => {
+      data.forEach(x => this.office.push(new Office(x)));
+    });
+    guestService.getGuest().subscribe(data => {
+      this.guest = data;
+    });
+    kitchenService.getKitchen().subscribe(data => {
+      data.forEach(x => this.kitchen.push(new Kitchen(x)));
+    });
+    printerService.getPrinter().subscribe(data => {
+      data.forEach(x => this.printer.push(new Printer(x)));
+    });
+    roomService.getRoom().subscribe(data => {
+      data.forEach(x => this.room.push(new Room(x)));
+    });
+
+    setTimeout(x => {
+      this.desk.forEach(y => {
+        this.all.push(new Desk(y));
+      });
+      this.kitchen.forEach(y => {
+        this.all.push(new Kitchen(y));
+      });
+      this.printer.forEach(y => {
+        this.all.push(new Printer(y));
+      });
+      this.office.forEach(y => {
+        this.all.push(new Office(y));
+      });
+      this.room.forEach(y => {
+        this.all.push(new Room(y));
+      });
+    }, 10000);
```

Comment:
Zakładam, że timeout z uwagi na brak danych, ale to niejest dobre rozwiązanie, bo powinno to pójść asynchronicznie, czyli jak załadują się dane to wtedy wrzucamy do `all`, jedyne co to przy takim rozwiązaniu trzeba się zastanowić kiedy, będą nam potrzebne dane podzielone, jeżeli i tak mamy je w `all`, wydaje się, że są niepotrzebne.

Używając rxjs, i spread operatorów ten konstruktor można uprościć, sprawdź czy działanie takie samo.
Ważne, żebyście przeczytali co robi `forkJoin` i `...`
```
    forkJoin(
      deskService.getDesk(),
      officeService.getOffice(),
      kitchenService.getKitchen(),
      printerService.getPrinter(),
      roomService.getRoom(),
      guestService.getGuest()
    ).subscribe(([desks, offices, kitchens, printers, rooms, guests]) => {
      this.desk = desks.map(x => new Desk(x));
      this.office = offices.map(x => new Office(x));
      this.kitchen = kitchens.map(x => new Kitchen(x));
      this.printer = printers.map(x => new Printer(x));
      this.room = rooms.map(x => new Room(x));
      this.guest = guests;

      this.all = [...this.desk, ...this.office, ...this.kitchen, ...this.printer, ...this.room];
    });
```

i zastanawiam się czy goście powinni tutaj trafić, bo raczej nie są powiązani z funkcjonalnością "clickables"